Documentation/articles/intro.md

```diff
@@ -1,5 +1,182 @@
+    Console.WriteLine($"Position: {parser.Latitude10000thMins / 60000.0},{parser.Longitude10000thMins / 60000.0}");
+}
+```
+
+Each of the parser types defines a property for each field in the corresponding AIS message type. For the most part
+these are pretty straightforward. The one potentially surprising feature is that none of the text fields returns a
+`string`. That's because the goal of this library is to minimize allocations, and to return a `string` it is usually
+necessary to allocate space for it on the GC heap. (The only exception is if the string can have one of a small set of
+known values, in which case you can return the same one every time for any particular value, instead of necessarily
+allocating a new one every time.) Text is instead represented using `NmeaAisTextFieldParser`.
+
+`NmeaAisTextFieldParser` lets you retrieve individual characters by index, or you can ask it to write the entire string
+in ASCII form into a `Span<byte>`. If you really want it as a `string` you can always do this:
+
+```csharp
+var parser = new NmeaAisStaticAndVoyageRelatedDataParser(
+    lineParser.Payload, lineParser.Padding);
+Span<byte> vesselNameAscii = stackalloc byte[(int)parser.VesselName.CharacterCount];
+parser.VesselName.WriteAsAscii(vesselNameAscii);
+// CAUTION: this will cause an allocation. Don't do this unless you have to
+string vesselName = Encoding.ASCII.GetString(vesselNameAscii);
+```
+
+By the way, the reason for copying the data into a separate `Span<byte>` first is that it is not stored in ASCII
+format in the AIS message. AIS uses its own 6-bit character encoding. The `NmeaAisTextFieldParser.WriteAsAscii`
+method converts from this into ASCII as it copies the data out.
+
+
+## Low-allocation design
+
+Ais.Net is designed to minimize GC overhead. As with any library there is some initial memory
+overhead to pay simply to use the code, but once you are up and running it is possible to
+process messages with no per-message allocations.
+
+The benchmarks built into the library source repository include a test that extracts location
+information from a file containing 1 million messages, and this has exactly the same memory
+allocation characteristics as a test that reads 1,000, or 10 million messages, demonstrating
+that after the first use, we make zero allocations per-message.
+
+Ais.Net achieves this by taking advantage of `Span<T>` and related types which were added in
+.NET Core 2.1, and using associated language features added in C# 7.2. This makes Ais.Net
+somewhat different to use from other .NET AIS parsers.
+
+The `Span<T>` and `ReadOnlySpan<T>` types impose a constraint in exchange for the high
+performance they make possible: these types can only live on the stack. This prevents variables
+of these types from living in fields of other types unless those containing types are
+defined in the same way. (So if you define a type that has the same constraint—if you define
```

Comment:
This seems a slightly odd sentence.

Reply:
I've rewritten this bit completely. So much so that if you still don't like it, it's probably best to start a new comment against the newly modified bit.